mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,688 @@
+    pub network: BlockchainNetwork,
+    /// The best block height.
+    pub best_block_height: AtomicU64,
+    /// Default fees to and confirmation targets to be used for FeeEstimator. Default fees are used when the call for
+    /// estimate_fee_sat fails.
+    pub default_fees_and_confirmations: PlatformCoinConfirmations,
+    /// This cache stores the transactions that the LN node has interest in.
+    pub registered_txs: PaMutex<HashMap<Txid, HashSet<Script>>>,
+    /// This cache stores the outputs that the LN node has interest in.
+    pub registered_outputs: PaMutex<Vec<WatchedOutput>>,
+    /// This cache stores transactions to be broadcasted once the other node accepts the channel
+    pub unsigned_funding_txs: PaMutex<HashMap<u64, TransactionInputSigner>>,
+}
+
+impl Platform {
+    pub fn new(
+        coin: UtxoStandardCoin,
+        network: BlockchainNetwork,
+        default_fees_and_confirmations: PlatformCoinConfirmations,
+    ) -> Self {
+        Platform {
+            coin,
+            network,
+            best_block_height: AtomicU64::new(0),
+            default_fees_and_confirmations,
+            registered_txs: PaMutex::new(HashMap::new()),
+            registered_outputs: PaMutex::new(Vec::new()),
+            unsigned_funding_txs: PaMutex::new(HashMap::new()),
+        }
+    }
+
+    pub fn update_best_block_height(&self, new_height: u64) {
+        self.best_block_height.store(new_height, AtomicOrdering::Relaxed);
+    }
+
+    pub fn best_block_height(&self) -> u64 { self.best_block_height.load(AtomicOrdering::Relaxed) }
+
+    pub fn add_tx(&self, txid: &Txid, script_pubkey: &Script) {
+        let mut registered_txs = self.registered_txs.lock();
+        match registered_txs.get_mut(txid) {
+            Some(h) => {
+                h.insert(script_pubkey.clone());
+            },
+            None => {
+                let mut script_pubkeys = HashSet::new();
+                script_pubkeys.insert(script_pubkey.clone());
+                registered_txs.insert(*txid, script_pubkeys);
+            },
+        }
+    }
+
+    pub fn add_output(&self, output: WatchedOutput) {
+        let mut registered_outputs = self.registered_outputs.lock();
+        registered_outputs.push(output);
+    }
+
+    async fn check_if_tx_is_onchain(&self, txid: Txid) -> Result<bool, MmError<UtxoRpcError>> {
+        if let Err(err) = self
+            .coin
+            .as_ref()
+            .rpc_client
+            .get_transaction_bytes(&H256Json::from(txid.as_hash().into_inner()).reversed())
+            .compat()
+            .await
+            .map_err(|e| e.into_inner())
+        {
+            if let UtxoRpcError::ResponseParseError(ref json_err) = err {
+                if let JsonRpcErrorType::Response(_, json) = &json_err.error {
+                    if let Some(message) = json["message"].as_str() {
+                        if message.contains("'code': -5") {
```

Comment:
Could you please make sure this is the only way to determine such case? Something can be changed on electrum side and it breaks mm2.

Reply:
Already answered in a previous review here 🙂 https://github.com/KomodoPlatform/atomicDEX-API/pull/1133#discussion_r763970045